apps/api/src/app/user/usecases/base-user-profile.usecase.ts

```diff
@@ -0,0 +1,36 @@
+import type { UserEntity } from '@novu/dal';
+import { UserResponseDto } from '../dtos/user-response.dto';
+
+export class BaseUserProfileUsecase {
+  protected mapToDto(user: UserEntity): UserResponseDto {
+    const {
+      _id,
+      resetToken,
+      resetTokenDate,
+      firstName,
+      lastName,
+      email,
+      profilePicture,
+      createdAt,
+      showOnBoarding,
+      servicesHashes,
+      jobTitle,
+      password,
+    } = user;
+
+    return {
+      _id,
+      resetToken,
+      resetTokenDate,
+      firstName,
+      lastName,
+      email,
+      profilePicture,
+      createdAt,
+      showOnBoarding,
+      servicesHashes,
+      jobTitle,
+      hasPassword: !!password,
+    };
+  }
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** Alternative, though not particularly important!‏
```suggestion
  protected mapToDto({ password, /* other exclusions */ ...userWithoutPw }: UserEntity): UserResponseDto {

    return {
      ...userWithoutPw,
      hasPassword: !!password,
    };
  }
```

Reply:
The thing is that with this approach whenever you add something to the `UserEntity` it will be automatically available in the `UserResponseDto` even if the fields are not defined in the DTO, with this entity it's better to control which fields are exposed.

Reply:
Very good call given the use case! Thanks for the explanation!